### PR TITLE
Feat/auth

### PR DESCRIPTION
**Pull request by jakgrab**
- Create LoginActivity with LoginFragment and RegistrationFragment,
- Set up Firebase auth for email and password user authentication

### CODE REVIEW

File: app/src/main/AndroidManifest.xml [hunk 1]
```diff
@@ -16,17 +16,22 @@
...
                 <action android:name="android.intent.action.MAIN" />
 
                 <category android:name="android.intent.category.LAUNCHER" />
             </intent-filter>
         </activity>
+        <activity
+            android:name="com.example.feature_main.MainActivity"
+            android:windowSoftInputMode="adjustResize">
+<!--            android:theme="@style/Theme.MarvelComics.MySplash">-->
+        </activity>
```
Comment (bibiosm98) [inline app/src/main/AndroidManifest.xml]: ?

File: core/src/main/java/com/example/core/repository/firebase_repository/FirebaseRepositoryImpl.kt [hunk 2]
```diff
@@ -0,0 +1,44 @@
+package com.example.core.repository.firebase_repository
+
+import android.util.Log
+import com.google.firebase.auth.FirebaseAuth
```
Comment (bibiosm98) [inline core/src/main/java/com/example/core/repository/firebase_repository/FirebaseRepositoryImpl.kt]: Are the logs still required?

File: feature-auth/src/main/res/layout/fragment_login.xml [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+        app:layout_constraintStart_toStartOf="parent"
+        app:layout_constraintTop_toBottomOf="@+id/etEmail" />
+
+    <Button
+        android:id="@+id/btSignIn"
+        android:layout_width="wrap_content"
+        android:layout_height="wrap_content"
+        android:layout_marginBottom="300dp"
+        android:text="@string/sign_in"
+        android:layout_marginVertical="20dp"
```
Comment (bibiosm98) [inline feature-auth/src/main/res/layout/fragment_login.xml]: <TextInputLayout>
    <TextInputEditText/>
 </TextInputLayout>
 
 Remove huge margins top/bottom, use constraints instead

File: feature-auth/src/main/res/layout/fragment_registration.xml [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+        android:id="@+id/btCreateAccount"
+        android:layout_width="wrap_content"
+        android:layout_height="wrap_content"
+        android:layout_marginBottom="300dp"
+        android:text="@string/create_account"
+        app:layout_constraintBottom_toBottomOf="parent"
+        app:layout_constraintEnd_toEndOf="parent"
+        app:layout_constraintHorizontal_bias="0.498"
+        app:layout_constraintStart_toStartOf="parent"
+        app:layout_constraintTop_toBottomOf="@+id/etPassword" />
```
Comment (bibiosm98) [inline feature-auth/src/main/res/layout/fragment_registration.xml]: Same

File: core/src/main/java/com/example/core/repository/firebase_repository/FirebaseRepositoryImpl.kt (hunk 2 above)
Comment (jakgrab) [reply core/src/main/java/com/example/core/repository/firebase_repository/FirebaseRepositoryImpl.kt]: Done

File: app/src/main/AndroidManifest.xml (hunk 1 above)
Comment (jakgrab) [reply app/src/main/AndroidManifest.xml]: Done

File: feature-auth/src/main/res/layout/fragment_login.xml (hunk 3 above)
Comment (jakgrab) [reply feature-auth/src/main/res/layout/fragment_login.xml]: Changed

File: feature-auth/src/main/res/layout/fragment_registration.xml (hunk 4 above)
Comment (jakgrab) [reply feature-auth/src/main/res/layout/fragment_registration.xml]: Changed

File: feature-auth/build.gradle [hunk 5]
```diff
@@ -0,0 +1,70 @@
...
+        dataBinding true
+//        viewBinding true
+    }
+}
+
+dependencies {
+    implementation project(':core')
+    //implementation project(':feature-main')
+
+    implementation 'androidx.core:core-ktx:1.9.0'
```
Comment (bibiosm98) [inline feature-auth/build.gradle:44]: comments

File: feature-auth/src/main/java/com/example/auth/LoginActivity.kt [hunk 6]
```diff
@@ -0,0 +1,63 @@
...
+        setContentView(binding.root)
+
+        auth = Firebase.auth
+
+        val navController = findNavController(R.id.nav_host_fragment_content_login)
+    }
+
+    override fun onStart() {
+        super.onStart()
+        val currentUser = auth.currentUser
```
Comment (bibiosm98) [inline feature-auth/src/main/java/com/example/auth/LoginActivity.kt:46]: is that worth to create variable if you use auth only there?

        val currentUser = Firebase.auth.currentUser instead

File: feature-auth/src/main/java/com/example/auth/fragments/registration/RegistrationFragment.kt [hunk 7]
```diff
@@ -0,0 +1,76 @@
...
+    // onDestroyView.
+    private val binding get() = _binding!!
+
+    private val registrationViewModel by viewModels<RegistrationViewModel>()
+
+    override fun onCreateView(
+        inflater: LayoutInflater, container: ViewGroup?,
+        savedInstanceState: Bundle?
+    ): View {
+//        _binding = FragmentRegistrationBinding.inflate(inflater, container, false)
```
Comment (bibiosm98) [inline feature-auth/src/main/java/com/example/auth/fragments/registration/RegistrationFragment.kt:34]: :(

File: feature-auth/src/main/res/layout/fragment_registration.xml [hunk 8]
```diff
@@ -0,0 +1,154 @@
...
+        <!--            style="@style/RegisterButtonStyle"-->
+        <!--            android:layout_width="match_parent"-->
+        <!--            android:layout_height="60dp"-->
+        <!--            android:layout_marginVertical="20dp"-->
+        <!--            android:clickable="true"-->
+        <!--            android:text="@string/sign_in"-->
+        <!--            app:layout_constraintBottom_toBottomOf="parent"-->
+        <!--            app:layout_constraintEnd_toEndOf="parent"-->
+        <!--            app:layout_constraintStart_toStartOf="parent"-->
+        <!--             />-->
```
Comment (bibiosm98) [inline feature-auth/src/main/res/layout/fragment_registration.xml:150]: :(

**Review by bibiosm98 — APPROVED** [review]

**Pull request merged**